config/default.yml

```diff
@@ -4102,6 +4328,12 @@ Style/MinMax:
   Enabled: true
   VersionAdded: '0.50'
 
+Style/MinMaxComparison:
+  Description: 'Enforces the use of `max` or `min` instead of comparison for greater or less.'
+  Enabled: false
```

Comment:
As we understand it, [this cop](https://github.com/rubocop/rubocop/blob/master/lib/rubocop/cop/style/min_max_comparison.rb) is not against `<`, but against `a < b ? a : b`, and suggests to use `[a, b].min` instead.

Voting for enabling it. Noise should be little.

Reply:
Yes, this is exactly what the cop is about. See my PR description; I do not consider _every_ replacement of a comparison statement like this with `min` or `max` an improvement.
When `a` or `b` are longer variable names or expressions, I would consider a proper `if something < other` over an array structure just to say `min`.

Having said that, I have not checked to which extent the cop will complain.

Reply:
Any `a < b ? a : b` is just a manual implementation of `min`:

```
# Unnecessarily verbose
x = if something < other
  something
else
  other
end

# Concise
x = [something, other].min
```

I cannot image a case where re-implementing min or max can be considered an improvement. In addition, I cannot remember to have seen this pattern anywhere in the wild. 

Still voting for enabling it. Noise should be little.

Reply:
Skimming through the discussion I'd tend to enable the cop as well. @foobear can you provide an concrete example where the additional verbosity adds value?

Reply:
The majority of the involved people seem to want to enable this cop, so I'll suggest we'll enable it.

Reply:
I personally don't think that

```ruby
if computed_value < maximum_value
  computed_value
else
  maximum_value
end
```

instead of

```ruby
[computed_value, maximum_value].min
```

is actually worse code, or that `min` improves the existing code.

Of course `min` does exactly that and using `<` is the same. But in cases like this, variable names including "maximum" with `min` being called on the tuple ist more difficult to read, especially for inexperienced developers.

I consider approaches equally valid and don't see how the cop helps in improving code here.

However, I'm happy to follow the majority opinion and enable this cop.

Reply:
Following up, I recently encountered this in the wild:

```ruby
[offset.to_i, 0].max # Not below zero
```

Which very much shows the point I was trying to make. :joy: 

In this case, the developer ended up using `clamp` with an open range instead.

```ruby
offset.clamp(0..)
```

While a bit odd-looking, I think this is better. At least better than `max`. :wink: